File: library/src/main/java/me/proxer/library/api/apps/InternalApi.java
package me.proxer.library.api.apps;

import me.proxer.library.api.ProxerCall;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

import javax.annotation.ParametersAreNullableByDefault;

/**
 * @author Ruben Gees
 */
@ParametersAreNullableByDefault
public interface InternalApi {

    @FormUrlEncoded
    @POST("apps/errorlog")
    ProxerCall<Void> errorLog(@Field("id") String id,
                              @Field("message") String message,
                              @Field("anonym") Boolean anonym,
                              @Field("silent") Boolean silent);
}

File: library/src/main/kotlin/me/proxer/library/api/apps/InternalApi.kt
package me.proxer.library.api.apps

import me.proxer.library.ProxerCall
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

/**
 * @author Ruben Gees
 */
internal interface InternalApi {

    @FormUrlEncoded
    @POST("apps/errorlog")
    fun errorLog(
        @Field("id") id: String?,
        @Field("message") message: String?,
        @Field("anonym") anonym: Boolean?,
        @Field("silent") silent: Boolean?
    ): ProxerCall<Unit>
}
